FCWTAPI/CWTObject.cs

```diff
@@ -145,26 +165,17 @@ public void SplitRealAndImaginary(CWTComponent comp, out double[,]? realCwt, out
             {
                 throw new ArgumentNullException("CWT must be performed before operating on it");
             }
-            if (OutputCWT.GetLength(0) % 2 != 0)
+            int rows = OutputCWT.RealArray.GetLength(0);
+            int cols = OutputCWT.ImagArray.GetLength(1); 
+            double[,] output = new double[rows, cols]; 
+            for(int i = 0; i < rows; i++)
             {
-                throw new ArgumentException("Cannot extract Real and Imaginary components from a non complex CWT");
-            }
-            int originalIndex = 0;
-            int outputRowLength = OutputCWT.GetLength(0) / 2;
-            double[,] outputArray = new double[outputRowLength, OutputCWT.GetLength(1)];
-            for (int i = 0; i < outputRowLength; i++)
-            {
-                int realRowIndex = originalIndex;
-                int imagRowIndex = originalIndex + 1;
-                for (int j = 0; j < OutputCWT.GetLength(1); j++)
+                for(int j = 0; j < cols; j++)
                 {
-                    double modPoint = Math.Sqrt(OutputCWT[realRowIndex, j] * OutputCWT[realRowIndex, j] + OutputCWT[imagRowIndex, j] * OutputCWT[imagRowIndex, j]);
-                    outputArray[i , j] = modPoint;
+                    output[i, j] = Math.Sqrt(Math.Pow(OutputCWT.RealArray[i, j], 2) + Math.Pow(OutputCWT.ImagArray[i, j], 2)); 
                 }
-                originalIndex += 2;
-                
             }
-            return outputArray;
+            return output;
         }
         /// <summary>
```

Comment:
Can you add a comment in here stating that this method of calculating the phase may be not great so that we remember later down the road?